vpr/src/device/rr_graph/rr_graph_obj.cpp

```diff
@@ -0,0 +1,1753 @@
+    VTR_ASSERT_MSG(node_type(node) == CHANX || node_type(node) == CHANY, "Track number valid only for CHANX/CHANY RR nodes");
+
+    set_node_ptc_num(node, track_id);
+}
+
+void RRGraph::set_node_class_num(RRNodeId node, short class_id) {
+    VTR_ASSERT(valid_node_id(node));
+    VTR_ASSERT_MSG(node_type(node) == SOURCE || node_type(node) == SINK, "Class number valid only for SOURCE/SINK RR nodes");
+
+    set_node_ptc_num(node, class_id);
+}
+
+void RRGraph::set_node_cost_index(RRNodeId node, short cost_index) {
+    VTR_ASSERT(valid_node_id(node));
+    node_cost_indices_[node] = cost_index;
+}
+
+void RRGraph::set_node_direction(RRNodeId node, e_direction direction) {
+    VTR_ASSERT(valid_node_id(node));
+    VTR_ASSERT_MSG(node_type(node) == CHANX || node_type(node) == CHANY, "Direct can only be specified on CHANX/CNAY rr nodes");
+
+    node_directions_[node] = direction;
+}
+
+void RRGraph::set_node_side(RRNodeId node, e_side side) {
+    VTR_ASSERT(valid_node_id(node));
+    VTR_ASSERT_MSG(node_type(node) == IPIN || node_type(node) == OPIN, "Side can only be specified on IPIN/OPIN rr nodes");
+
+    node_sides_[node] = side;
+}
+
+void RRGraph::set_node_R(RRNodeId node, float R) {
+    VTR_ASSERT(valid_node_id(node));
+
+    node_Rs_[node] = R;
+}
+
+void RRGraph::set_node_C(RRNodeId node, float C) {
+    VTR_ASSERT(valid_node_id(node));
+
+    node_Cs_[node] = C;
+}
+
+/*
+ * Set a segment id for a node in rr_graph 
+ */
+void RRGraph::set_node_segment(RRNodeId node, RRSegmentId segment_id) {
+    VTR_ASSERT(valid_node_id(node));
+
+    /* Only CHANX and CHANY requires a valid segment id */
+    if ((CHANX == node_type(node))
+        || (CHANY == node_type(node))) {
+        VTR_ASSERT(valid_segment_id(segment_id));
+    }
+
+    node_segments_[node] = segment_id;
+
+    return;
+}
+
+/* For a given node in a rr_graph
+ * classify the edges of each node to be configurable (1st part) and non-configurable (2nd part) 
+ */
+void RRGraph::partition_node_in_edges(RRNodeId node) {
+    //Partition the edges so the first set of edges are all configurable, and the later are not
+    auto first_non_config_edge = std::partition(node_in_edges_[node].begin(), node_in_edges_[node].end(),
+                                                [&](const RREdgeId edge) { return edge_is_configurable(edge); }); /* Condition to partition edges */
+
+    size_t num_conf_edges = std::distance(node_in_edges_[node].begin(), first_non_config_edge);
+    size_t num_non_conf_edges = node_in_edges_[node].size() - num_conf_edges; //Note we calculate using the size_t to get full range
+
+    /* Check that within allowable range (no overflow when stored as num_non_configurable_edges_
+     */
+    VTR_ASSERT_MSG(num_non_conf_edges <= node_in_edges_[node].size(),
+                   "Exceeded RR node maximum number of non-configurable input edges");
+
+    node_num_non_configurable_in_edges_[node] = num_non_conf_edges; //Narrowing
+
+    return;
```

Comment:
Don't need return

Reply:
Removed for all the returns without values